SignalServiceKit/src/Loki/API/LokiGroupChatAPI.swift

```diff
@@ -209,6 +208,48 @@ public final class LokiGroupChatAPI : NSObject {
         }
     }
     
+    public static func deleteMessageWithServerID(_ messageServerID: UInt, for group: UInt64, on server: String, isOurOwnMessage: Bool = true) -> Promise<Void> {
+        return getAuthToken(for: server).then { token -> Promise<Void> in
+            let modTag = isOurOwnMessage ? "" : "[Mod]"
+            print("[Loki]\(modTag) Deleting message with server ID: \(messageServerID) for group chat with ID: \(group) on server: \(server).")
+        
+            let endpoint = isOurOwnMessage ? "\(server)/channels/\(group)/messages/\(messageServerID)" : "\(server)/loki/v1/moderation/message/\(messageServerID)"
+            let url = URL(string: endpoint)!
+            let request = TSRequest(url: url, method: "DELETE", parameters: [:])
+            request.allHTTPHeaderFields = [ "Content-Type" : "application/json", "Authorization" : "Bearer \(token)" ]
+            return TSNetworkManager.shared().makePromise(request: request).map { result -> Void in
```

Comment:
`map` is for transforming the promise result. `done` would be more appropriate here